src/request.ts

```diff
@@ -58,10 +51,21 @@ const newRequestFromIncoming = (
       headerRecord.push([key, value])
     }
   }
+  return new Headers(headerRecord)
```

Comment:
Using `new Headers` instead of creating `Request` is nice.

It's just an idea, maybe overwork, we should not add it in this PR, but is creating `LightweightHeaders` a good idea?

```ts
class LightweightHeaders {
  constructor(rawHeaders) {} // or constructor(headerRecord)

  get(name) {
    // find the value from rawHeaders or headerRecord
  }

  has(name) {
    // find the value from rawHeaders or headerRecord
  }

  set(name, value) {
    this.toHeaders.set(name, value)
  }

  toHeaders() {
    const headers = new Headers(this.entries())
    // cache headers
    return headers
  }
}
```

It's for readonly, so it returns the value by `.get()` from `rawHeaders` or `headerRecord`, and if accessing `.set()`, do `new Headers()`.

Anyway, we can merge this PR.

Reply:
`LightweightHeaders` is a great idea! However, I'm concerned that using it like `fetch(url, {headers: c.req.raw.headers})` would introduce more compatibility issues to worry about, making it not worth the maintenance cost.